src/main/java/org/folio/services/instance/InstanceService.java

```diff
@@ -64,7 +64,11 @@ public Future<Response> createInstance(Instance entity) {
           InstanceStorage.PostInstanceStorageInstancesResponse.class, postResponse);
 
         return postResponse.future()
-          .compose(domainEventPublisher.publishCreated());
+          .compose(response -> {
+            // return response without waiting for event publish
+            domainEventPublisher.publishCreated().apply(response);
+            return Future.succeededFuture(response);
+          });
```

Comment:
I thought the scope of this work was to change the database operations relating to HRIDs.

Why is this change also included in this pull request?

Reply:
After initial changes were made relating to the db and profiling the application, I saw that there was still room for improvement. The reason for the story is to optimize this hot spot for data import. I can push this out to another story if you feel strongly.

Reply:
> After initial changes were made relating to the db and profiling the application, I saw that there was still room for improvement. The reason for the story is to optimize this hot spot for data import.

Thank you for explaining that.

> I can push this out to another story if you feel strongly.

There is no need to split it into another issue. I only wanted to understand why this change was made as well.

I think it could be worth expanding the comment to explain why the message is published asynchronously.

Reply:
expand my comment.